test/e2e/interconnections/create_test.go

```diff
@@ -0,0 +1,97 @@
+package interconnections
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	pager "github.com/equinix/metal-cli/internal/pagination"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/interconnections"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/test/helper"
+
+	metal "github.com/equinix/equinix-sdk-go/services/metalv1"
+	"github.com/spf13/cobra"
+)
+
+func TestInterconnections_Create(t *testing.T) {
+	subCommand := "interconnections"
+	rootClient := root.NewClient(helper.ConsumerToken, helper.URL, helper.Version)
+	randomString := helper.GenerateRandomString(5)
+
+	project := helper.CreateTestProject(t, "metal-cli-interconnections-create-"+randomString)
+	vlan := helper.CreateTestVLAN(t, project.GetId())
+
+	apiClient := helper.TestClient()
+
+	tests := []struct {
+		name    string
+		cmd     *cobra.Command
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "create shared vlan interconnection",
+			cmd:  interconnections.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				connName := "conn-1-" + randomString
+
+				root.SetArgs([]string{subCommand, "create", "-p", project.GetId(), "--vlan", fmt.Sprintf("%d", vlan.GetVxlan()), "-n", connName, "-m", vlan.GetMetroCode(), "-r", "primary", "-t", "shared", "-T", "a_side", "-s", "50000000"})
+
+				out := helper.ExecuteAndCaptureOutput(t, root)
+
+				conns, err := pager.GetAllProjectInterconnections(*apiClient.InterconnectionsApi, project.GetId(), []string{}, []string{})
```

Comment:
Two notes here:

1. Based on what the portal is doing, we could instead get organization interconnections and specify the project ID as a request parameter in order to get only the interconnections that belong to the test project.
2. The `pager` functions should not be used any more; as of now, the SDK should be providing its own paginator functions (`ExecuteWithPagination` instead of `Execute`), and we should use those instead of making/maintaining our own in the CLI

Reply:
I tried using to get the connections for whole org but for project, when created using `default org` organization object is not populated in its model. Even after fetching it explicitly.

Also, for fetching all connections for an org, there is no pagination it seems as there is no `ExecuteWithPagination` nor `pager.GetAllOrganizationInterconnections()`

Added the fix to fetch default org of current user and use that ID to get conns for an org

Reply:
@ctreatma I have used `org connections` in the tests but still failing with the [same schema error](https://github.com/equinix/metal-cli/actions/runs/7558522078/job/20581385766?pr=397#step:5:111). The problem here is if there is any connection for a project or Org which was created with speed out of `int32` limit, is failing to get via metal-go.

My local tests were passing because I deleted all the other interconnection with more speeds.

Reply:
There's a draft PR up to work around this API issue within the SDK; could you try it out and confirm if it resolves your issue? https://github.com/equinix/equinix-sdk-go/pull/22

Reply:
The interconnection create test is failing but I suspect it's related to an ongoing incident because the error is a generic HTTP 500 error.  We'll have to wait until the incident is resolved and then re-run.  Once the patch is validated it can be merged and released, and this PR can be updated to use the latest SDK release.

Reply:
When I re-ran the test it still failed, but this time with a 422 error during cleanup, which indicates that the cleanup code is doing something it shouldn't do.  I've released v0.31.1 of equinix-sdk-go, so you can pull that in to your PR and that should resolve the `int32` errors, but you will still need to dig into the test cleanup failure.

Reply:
Yes, I'll check the cleanup failure issue